test/integration/controllermanager/event/event_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package event_test
+
+import (
+	"time"
+
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("Event controller tests", func() {
+	var (
+		shootEvent, nonShootEvent *corev1.Event
+
+		// For testing purpose we are setting this variable to be same as TTLNonShootEvents, to have
+		// timeUntilDeletion value in controller to be 0 or less that 0. This is done to mock the deletion of non-shoot events
+		// on reaching TTL.
+		ttl = &metav1.Duration{Duration: 30 * time.Minute}
+	)
+
+	BeforeEach(func() {
+		shootEvent = &corev1.Event{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: testID + "-",
+				Namespace:    testNamespace.Name,
+			},
+			LastTimestamp:  metav1.Time{Time: time.Now().Add(-ttl.Duration)},
+			InvolvedObject: corev1.ObjectReference{Kind: "Shoot", APIVersion: "core.gardener.cloud/v1beta1", Namespace: testNamespace.Name},
+		}
+
+		nonShootEvent = &corev1.Event{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: testID + "-",
+				Namespace:    testNamespace.Name,
+			},
+			LastTimestamp:  metav1.Time{Time: time.Now().Add(-ttl.Duration)},
+			InvolvedObject: corev1.ObjectReference{Kind: "Project", APIVersion: "core.gardener.cloud/v1beta1", Namespace: testNamespace.Name},
+		}
+	})
+
+	JustBeforeEach(func() {
```

Comment:
You could create the events only in the test cases where they are required instead of creating them always although not required.